src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -4,29 +4,197 @@
  */
 import * as React from 'react';
 import flow from 'lodash/flow';
-import { FormattedMessage } from 'react-intl';
+import getProp from 'lodash/get';
+import { FormattedMessage, type MessageDescriptor } from 'react-intl';
+
+import { Text } from '@box/blueprint-web';
+import { AddMetadataTemplateDropdown } from '@box/metadata-editor';
+
+import API from '../../api';
+import { type ElementsXhrError } from '../../common/types/api';
+import { type ElementOrigin } from '../common/flowTypes';
+import { isUserCorrectableError } from '../../utils/error';
 import { withAPIContext } from '../common/api-context';
 import { withErrorBoundary } from '../common/error-boundary';
 import { withLogger } from '../common/logger';
-import { ORIGIN_METADATA_SIDEBAR_REDESIGN } from '../../constants';
+import {
+    FIELD_IS_EXTERNALLY_OWNED,
+    FIELD_PERMISSIONS,
+    FIELD_PERMISSIONS_CAN_UPLOAD,
+    IS_ERROR_DISPLAYED,
+    ORIGIN_METADATA_SIDEBAR_REDESIGN,
+} from '../../constants';
 import { EVENT_JS_READY } from '../common/logger/constants';
 import { mark } from '../../utils/performance';
+import LoadingIndicator from '../../components/loading-indicator/LoadingIndicator';
+import InlineError from '../../components/inline-error/InlineError';
+
 import messages from '../common/messages';
 
+import { type BoxItem } from '../../common/types/core';
+import { MetadataTemplate, type MetadataEditor } from '../../common/types/metadata';
+import { type WithLoggerProps } from '../../common/types/logging';
+
 import './MetadataSidebarRedesign.scss';
 
 const MARK_NAME_JS_READY = `${ORIGIN_METADATA_SIDEBAR_REDESIGN}_${EVENT_JS_READY}`;
 
 mark(MARK_NAME_JS_READY);
 
-function MetadataSidebarRedesign() {
+interface ExternalProps {
+    isFeatureEnabled: boolean;
+}
+
+interface PropsWithoutContext extends ExternalProps {
+    elementId: string;
+    fileId: string;
+    hasSidebarInitialized?: boolean;
+}
+
+interface ErrorContextProps {
+    onError: (error: ElementsXhrError | Error, code: string, contextInfo?: Object, origin?: ElementOrigin) => void;
+}
+
+export interface MetadataSidebarRedesignProps extends PropsWithoutContext, ErrorContextProps, WithLoggerProps {
+    api: API;
+}
+
+type Status = 'idle' | 'loading' | 'error' | 'success';
+interface DataFetcher {
+    status: Status;
+    file: BoxItem | null;
+    errorMessage: MessageDescriptor | null;
+    templates: Array<MetadataTemplate>;
+}
+
+function useDataFetching(
```

Comment:
wondering if this hook could be defined in separate file 🤔

Reply:
Also the hook is tied with fetching metadata, think something like `useMetadataFetcher` or `useSidebarMetadataFetcher` would work

Reply:
Ok I moved the hook into /hooks folder